pkgs/development/libraries/microsoft-wil/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, gtest
+, pkg-config
+}:
+
+stdenv.mkDerivation rec {
+  pname = "microsoft-wil";
+  version = "1.0.230411.1";
+
+  src = fetchFromGitHub {
+    owner = "Microsoft";
+    repo = "wil";
+    rev = "v${version}";
+    hash = "sha256-5DBJDgwZljfiCFMPlPE9aZ41rKnEVzLUcxRTgxqPWHw=";
+  };
+
+  nativeBuildInputs = [ cmake pkg-config ];
+  buildInputs = [ gtest ];
+
+  cmakeFlags = [
+    "-DWIL_BUILD_PACKAGING=OFF"
+    # Unit tests require downloads via nuget
+    "-DWIL_BUILD_TESTS=OFF"
+  ];
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "Windows Implementation Library";
+    longDescription = ''
+    A header-only C++ library created to make life easier for developers on Windows through readable type-safe C++ interfaces for common Windows coding patterns.
```

Comment:
```suggestion
      A header-only C++ library created to make life easier for developers on Windows through readable type-safe C++ interfaces for common Windows coding patterns.
```

Reply:
Is it ok to vendor `microsoft-wil` for the time being? `onnxruntime` seems to be the only package in `nixpkgs` to depend on it. Given the track record of `onnxruntime` it seems rather likely that it will not stay up-to-date with the latest upstream version of `microsoft-wil` anyhow.